src/stage1/zigendian.h

```diff
@@ -0,0 +1,34 @@
+#ifndef ZIG_ENDIAN_H
+#define ZIG_ENDIAN_H
+
+// Every OSes seem to define endianness macros in different files.
```

Comment:
Maybe move this in `utils_base.hpp` instead of a new file?

Reply:
It's also used in parse_f128.c, would it be fine to move it to a C++ header file?